spot-contracts/contracts/strategies/FeeStrategy.sol

```diff
@@ -0,0 +1,227 @@
+        maxBurnFeePerc = maxBurnFeePerc_;
+    }
+
+    /// @notice Update the parameters determining the slope and asymptotes of the sigmoid fee curve.
+    /// @param p Lower, Upper and Growth sigmoid paramters are fixed point numbers with {PERC_DECIMALS} places.
+    function updateRolloverFees(SigmoidParams memory p) external onlyOwner {
+        require(p.lower >= -int256(MAX_PERC), "FeeStrategy: fee bound too low");
+        require(p.upper <= int256(MAX_PERC), "FeeStrategy: fee bound too high");
+        require(p.lower < p.upper, "FeeStrategy: paramters invalid");
+        rolloverFeeAPR.lower = p.lower;
+        rolloverFeeAPR.upper = p.upper;
+        rolloverFeeAPR.growth = p.growth;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function computeMintFeePerc() external override returns (uint256) {
+        // when vault tvl > target, we encorage supply growth by dropping mint fees to 0
+        return (computeDeviationRatio(perp.getDepositBond()) > HUNDRED_PERC) ? 0 : maxMintFeePerc;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function computeBurnFeePerc() external override returns (uint256) {
+        // when vault tvl < target, we encorage supply reduction by dropping burn fees to 0
+        return (computeDeviationRatio(perp.getDepositBond()) < HUNDRED_PERC) ? 0 : maxBurnFeePerc;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function computeRolloverFeePerc() external override returns (int256) {
+        IBondController referenceBond = perp.getDepositBond();
+        int256 rolloverAPR = Sigmoid.compute(
+            computeDeviationRatio(referenceBond).toInt256(),
+            rolloverFeeAPR.lower,
+            rolloverFeeAPR.upper,
+            rolloverFeeAPR.growth,
+            PERC_DECIMALS
+        );
+
+        // We calculate the rollover fee for the given cycle by dividing the annualized rate
+        // by the number of cycles in any given year.
+        return (rolloverAPR * referenceBond.duration().toInt256()) / ONE_YEAR_SEC;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function decimals() external pure override returns (uint8) {
+        return PERC_DECIMALS;
+    }
+
+    /// @notice Computes the ratio between the vault's TVL and the target TVL.
+    /// @dev Adjusts the computed result based on the `deviationThresholdPerc`.
+    /// @return The ratio as a fixed point number with {PERC_DECIMALS} places.
+    function computeDeviationRatio(IBondController referenceBond) public returns (uint256) {
+        // NOTE: Ensure that the perp's TVL and vault's TVL have the same base denomination.
+        (uint256 perpRatio, ) = perp.computeEffectiveTrancheRatio(referenceBond);
+        uint256 targetTVL = perp.getTVL().mulDiv(TRANCHE_RATIO_GRANULARITY, perpRatio);
+        uint256 deviationFactor = vault.getTVL().mulDiv(HUNDRED_PERC, targetTVL);
+
+        // For example, the threshold is set to 5%.
+        // When the computed ratio is below 1, we leave it unchanged.
```

Comment:
Not sure I follow this logic.

So, 5% is not +/-5%?
I.e. If it's 0.97, should it up moved up to 1.0?

I'd expect (0.95, 1.05) -> 1.0, otherwise, leave it unchanged.

Reply:
I think in the context of the rotation vault, 0.97 isn't good enough as it wont rollover the entire supply.
What we really want is 1.0 or more, thus the neutral band is defined as (1.0 - 1.05) ..